src/common/link.rs

```diff
@@ -109,135 +110,149 @@ pub struct LinkValue {
-    /// service.
-    Service,
-    /// start.
-    Start,
-    /// stylesheet.
-    Stylesheet,
-    /// subsection.
-    Subsection,
-    /// successor-version.
-    SuccessorVersion,
-    /// up.
-    Up,
-    /// versionHistory.
-    VersionHistory,
-    /// via.
-    Via,
-    /// working-copy.
-    WorkingCopy,
-    /// working-copy-of.
-    WorkingCopyOf,
-    /// ext-rel-type.
-    ExtRelType(String)
+
+impl_variants! {
+    /// A Media Descriptor, based on:
+    /// [https://www.w3.org/TR/html401/types.html#h-6.13][url]
+    ///
+    /// [url]: https://www.w3.org/TR/html401/types.html#h-6.13
+    name: MediaDesc,
+    mod_name: media_desc,
+
+    SCREEN => "screen",
+    TTY => "tty",
+    TV => "tv",
+    PROJECTION => "projection",
+    HANDHELD => "handheld",
+    PRINT => "print",
+    BRAILLE => "braille",
+    AURAL => "aural",
+    ALL => "all",
+}
+
+impl_variants! {
+    /// A Link Relation Type, based on:
+    /// [RFC5988](https://tools.ietf.org/html/rfc5988#section-6.2.2)
+    name: RelationType,
+    mod_name: relation_type,
+
+    ALTERNATE => "alternate",
+    APPENDIX => "appendix",
+    BOOKMARK => "bookmark",
+    CHAPTER => "chapter",
+    CONTENTS => "contents",
+    COPYRIGHT => "copyright",
+    CURRENT => "current",
+    DESCRIBED_BY => "described-by",
+    EDIT => "edit",
+    EDIT_MEDIA => "edit-media",
+    ENCLOSURE => "enclosure",
+    FIRST => "first",
+    GLOSSARY => "glossary",
+    HELP => "help",
+    HUB => "hub",
+    INDEX => "index",
+    LAST => "last",
+    LATEST_VERSION => "latest-version",
+    LICENSE => "license",
+    NEXT => "next",
+    NEXT_ARCHIVE => "next-archive",
+    PAYMENT => "payment",
+    PREV => "prev",
+    PREDECESSOR_VERSION => "predecessor-version",
+    PREVIOUS => "previous",
+    PREV_ARCHIVE => "prev-archive",
+    RELATED => "related",
+    REPLIES => "replies",
+    SECTION => "section",
+    SELF => "self",
+    SERVICE => "service",
+    START => "start",
+    STYLESHEET => "stylesheet",
+    SUBSECTION => "subsection",
+    SUCCESSOR_VERSION => "successor-version",
+    UP => "up",
+    VERSION_HISTORY => "version-history",
+    VIA => "via",
+    WORKING_COPY => "working-copy",
+    WORKING_COPY_OF => "working-copy-of",
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 // Struct methods
 ////////////////////////////////////////////////////////////////////////////////
 
 impl Link {
-    /// Create `Link` from a `Vec<LinkValue>`.
-    pub fn new(link_values: Vec<LinkValue>) -> Link {
-        Link { values: link_values }
+    /// Create `Link` with an empty list of values.
+    pub fn new() -> Link {
+        Self::default()
     }
 
-    /// Get the `Link` header's `LinkValue`s.
-    pub fn values(&self) -> &[LinkValue] {
-        self.values.as_ref()
+    /// Get an iterator over the `Link` header's `LinkValue`s.
+    pub fn iter(&self) -> impl Iterator<Item = LinkValue> + '_ {
+        self.values.iter().cloned()
```

Comment:
Should the default iterator be cloning the values? Why not give references, and a user can clone if they need to?

Reply:
Do you want it changed?